File: rl/policies/sensors/Sensor.py
#The gol of a sensor is to produce an information that can be used by a policy
from rl.policies.sensors.SensorError import SensorError

class Sensor:

    def process(self,observation):
        raise SensorError("No process function defined")

    def size(self):
        raise SensorError("No size defined")

File: torch/core/Sensor.lua
 
 --- The gaol of a sensor is to transform an observation to a data structure that can be processed by a policy
 local Sensor = torch.class('rltorch.Sensor'); 

function Sensor:__init(observation_space)
  self.observation_space=observation_space
end

function Sensor:process(observation)
  assert(false,"No process function")
end

